### DISCUSSION TITLE
Is Vite supported only for component testing?

### DISCUSSION THREAD

**Discussion by mrmartineau** [cypress-io/cypress#18017] [General]
Can anyone confirm that Vite is supported only for component testing? I ask because the only mention of it on the Cypress website is in relation to component tests.

I have [followed the instructions](https://docs.cypress.io/guides/component-testing/framework-configuration#Vite-Based-Projects-Vue-React) and have **successfully setup component tests** for my Vite-based app, but I would also like to run end-to-end tests for the app too. Is this currently possible?

FYI, here are the npm scripts that I use to run either the component tests or the e2e tests:

```json
{
  "test:cypress:e2e": "cypress open",
  "test:cypress:e2e:ci": "start-server-and-test 'yarn serve-vite' http://localhost:5000 'cypress run-ct --config baseUrl=http://localhost:5000'",
  "test:cypress:components": "cypress open-ct",
  "test:cypress:components:ci": "cypress run-ct --quiet --reporter spec",
}
```

And my `cypress/plugins/index.js` file:

```js
/**
 * @type {Cypress.PluginConfig}
 */

const path = require('path')
const { startDevServer } = require('@cypress/vite-dev-server')

module.exports = (on, config) => {
  // `on` is used to hook into various events Cypress emits
  // `config` is the resolved Cypress config
  on('dev-server:start', (options) => {
    return startDevServer({
      options,
      viteConfig: {
        configFile: path.resolve(__dirname, '..', '..', 'vite.config.ts'),
      },
    })
  })
}
```

Any ideas?

**Comment by herecydev** [comment]
Absolutely, cypress isn't tied to technology, anything that runs in the browser should be able to be tested

**Comment by jennifer-shehane** [comment]
Just need the url of the application to visit and Cypress end to end can test anything there. `cy.visit(url)`
